File: BookNestBackend/BookNest.Services/Database/Entities/Organizer.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BookNest.Services.Database.Entities
{
    public class Organizer
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(50)]
        public string FirstName { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string LastName { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        [EmailAddress]
        public string ContactEmail { get; set; } = string.Empty;

        [Phone]
        public string? PhoneNumber { get; set; } 

        public ICollection<Event> Events { get; set; } = new List<Event>();
        
    }
}

File: BookNestFrontend/booknest_desktop/lib/models/organizer.dart
import 'event.dart';

class Organizer {
  final int id;
  final String firstName;
  final String lastName;
  final String contactEmail;
  final String? phoneNumber;
  final List<Event> events;

  Organizer({
    required this.id,
    required this.firstName,
    required this.lastName,
    required this.contactEmail,
    this.phoneNumber,
    this.events = const [],
  });

  String get name => '$firstName $lastName'.trim();
  int get eventCount => events.length;

  factory Organizer.fromJson(Map<String, dynamic> json) {
    return Organizer(
      id: json['id'],
      firstName: json['firstName'] ?? '',
      lastName: json['lastName'] ?? '',
      contactEmail: json['contactEmail'] ?? '',
      phoneNumber: json['phoneNumber'],
      events: (json['events'] as List<dynamic>? ?? [])
          .map((e) => Event.fromJson(e))
          .toList(),
    );
  }
}
